### ISSUE TITLE
libssl.so.1.1: cannot open shared object file

### ISSUE THREAD

**Issue by ShiroYasha139** [phil294/AHK_X11#18]
When I run ahk, I get the above error in console. I did check #9 but I'm not running Wayland. I'm running X11 with EndeavourOS (KDE). Also how do I uninstall ahk_x11? I messed up something by removing it from application preference.

**Comment by phil294** [comment]
ok I will look into it.

>  I messed up something by removing it from application preference.

In this case probably the easiest fix is to reinstall as it overwrites the Desktop files etc.

but if you really want to

> uninstall ahk_x11

you need to revert the steps taken in https://github.com/phil294/AHK_X11/blob/master/src/installer.ahk. As far as I can see, this boils down to:

```bash
#!/bin/bash
set -e
rm ~/.local/bin/ahk_x11
xdg-icon-resource uninstall --context mimetypes --size 48 application-x-ahk_x11
xdg-mime uninstall ~/.local/share/mime/application/x-ahk_x11.xml
update-mime-database ~/.local/share/mime
rm ~/.local/share/applications/ahk_x11.desktop
rm ~/.local/share/applications/ahk_x11-compiler.desktop
update-desktop-database ~/.local/share/applications
```
seems right, but please tell me if it works for you too.

Eventually, I'd like to have a proper AUR package for Arch-based distros which you can then just uninstall. I'm already talking to a volunteer for that, just need to fix the installation process from root's point of view.

...and similar for other distros too of course

**Comment by ShiroYasha139** [comment]
Arch Linux recently updated from openssl v1.1 to v3 i think so that's the cause. I just remembered I keep seeing issues about it on r/ArchLinux. It's probably due to that. Installing the older version alongside it worked.

Thank you for the uninstallation procedure. I don't see it in application preference order after using the commands.

**Comment by ShiroYasha139** [comment]
Arch Linux recently updated from openssl v1.1 to v3 i think so that's the cause. I just remembered I keep seeing issues about it on r/ArchLinux. It's probably due to that. Installing the older version alongside it worked.

Thank you for the uninstallation procedure. I don't see it in application preference order after using the commands.

**Comment by phil294** [comment]
thx for the investigation. All libraries that aren't likely eternally stable should be statically linked to the library. This is important for cross platform compatibility for compiled scripts. Seems I overlooked some libraries in the build script... I'll fix it for the next release, which may still be a few weeks away.

**Comment by phil294** [comment]
closed with 9c0de54d
